libbeat/beat/pipeline.go

```diff
@@ -136,7 +136,6 @@ type ClientListener interface {
 	Closed()  // Closed indicates the client being fully shutdown
 
 	Published()             // event has successfully entered the queue
-	FilteredOut(Event)      // event has been filtered out/dropped by processors
```

Comment:
PR looks fine, but conceptually the existence of this makes sense. If we need to ack an event from the Filebeat shipper back to the originating input, but the event was in fact dropped by a processor, do we have a way to ack in this case?

Is this just duplicated by the `DroppedOnPublish` processor?

I want to make sure that this not being used isn't a hint that one of our acknowledgement edge cases won't be covered, although I think this would be a problem for Filebeat registry updates today regardless.

Reply:
No, this is in every way equivalent to calling the `EventListener.AddEvent` callback with the `published` parameter set to `false` (`DroppedOnPublish` has a different meaning and is called almost exclusively after the client has been closed). `EventListener.AddEvent(_, false)` and `ClientListener.FilteredOut(_)` are called by the same code path at the same time, and the `ClientListener` and `EventListener` are both set in parameters to the same `pipeline.ConnectWith` call, so anyone that can provide one can easily provide the other. It just happens that everyone who has needed this notification has implemented it by checking for `published==false` on `AddEvent`, so imo having this unused equivalent makes the overall state transition flow more confusing. (My larger hope is to document and possibly coalesce the two partially-overlapping callback structs, but this cleanup is all I had time for today 😅)

Reply:
Thanks for the explanation!